### PR TITLE
Implement missing Webview features on iOS (GSoC 2023)

### PR DESCRIPTION
**Pull request by Codnelle**

### CODE REVIEW

**Comment by AppInventorWorkerBee** [comment]
Can one of the admins verify this patch?

**Comment by ewpatton** [comment]
@Codnelle We will need to rebase this to ucr and also so we can remove the ListView related changes.

File: appinventor/AICompanionApp.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -349,8 +349,9 @@
 				TargetAttributes = {
 					06463EF71D95956800B23DCD = {
 						CreatedOnToolsVersion = 8.0;
+						DevelopmentTeam = 3FKTLL5XPW;
 						LastSwiftMigration = 1100;
-						ProvisioningStyle = Manual;
+						ProvisioningStyle = Automatic;
```
Comment (jisqyv) [inline appinventor/AICompanionApp.xcodeproj/project.pbxproj:354]: Is this appropriate?

File: appinventor/AICompanionApp.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -786,18 +787,20 @@
...
+				DEVELOPMENT_TEAM = 3FKTLL5XPW;
 				FRAMEWORK_SEARCH_PATHS = (
 					"$(inherited)",
 					"$(PROJECT_DIR)/prebuilts/$(PLATFORM_NAME)",
 				);
 				INFOPLIST_FILE = aicompanionapp/src/Info.plist;
 				IPHONEOS_DEPLOYMENT_TARGET = 12.0;
 				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks";
-				PRODUCT_BUNDLE_IDENTIFIER = edu.mit.appinventor.aicompanion3;
+				PRODUCT_BUNDLE_IDENTIFIER = com.kashishsharma.me;
```
Comment (jisqyv) [inline appinventor/AICompanionApp.xcodeproj/project.pbxproj:800]: Need to reverse this change

File: appinventor/AICompanionApp.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -786,18 +787,20 @@
...
 				PRODUCT_NAME = "$(TARGET_NAME)";
 				PROVISIONING_PROFILE = "9c96ca04-1d70-4a2b-bb01-28f545b35d28";
-				PROVISIONING_PROFILE_SPECIFIER = "MIT App Inventor Development";
-				"PROVISIONING_PROFILE_SPECIFIER[sdk=iphoneos*]" = "MIT App Inventor Development";
+				PROVISIONING_PROFILE_SPECIFIER = "";
```
Comment (jisqyv) [inline appinventor/AICompanionApp.xcodeproj/project.pbxproj:803]: Same here I suspect

File: appinventor/AICompanionApp.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -808,18 +811,20 @@
...
+				CODE_SIGN_STYLE = Automatic;
+				DEVELOPMENT_TEAM = 3FKTLL5XPW;
 				FRAMEWORK_SEARCH_PATHS = (
 					"$(inherited)",
 					"$(PROJECT_DIR)/prebuilts/$(PLATFORM_NAME)",
 				);
 				INFOPLIST_FILE = aicompanionapp/src/Info.plist;
 				IPHONEOS_DEPLOYMENT_TARGET = 12.0;
 				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks";
-				PRODUCT_BUNDLE_IDENTIFIER = edu.mit.appinventor.aicompanion3;
```
Comment (jisqyv) [inline appinventor/AICompanionApp.xcodeproj/project.pbxproj:818]: And here

File: appinventor/AICompanionApp.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -808,18 +811,20 @@
...
+				PRODUCT_BUNDLE_IDENTIFIER = com.kashishsharma.me;
 				PRODUCT_NAME = "$(TARGET_NAME)";
 				PROVISIONING_PROFILE = "9c96ca04-1d70-4a2b-bb01-28f545b35d28";
-				PROVISIONING_PROFILE_SPECIFIER = "MIT App Inventor Production";
-				"PROVISIONING_PROFILE_SPECIFIER[sdk=iphoneos*]" = "MIT App Inventor Production";
```
Comment (jisqyv) [inline appinventor/AICompanionApp.xcodeproj/project.pbxproj:822]: And here

**Comment by ewpatton** [comment]
@daki7711 @Codnelle We need to make sure to rebase the changes onto ucr and update the base branch appropriately here. This should not be merged directly into master.

**Review by ewpatton — APPROVED** [review]

**Pull request merged**